src/metabase/models/database.clj

```diff
@@ -302,6 +302,14 @@
   "The string to replace passwords with when serializing Databases."
   "**MetabasePass**")
 
+(def ^:const protected-db-details
+  "The map to replace details with when serializing Databases."
+  {:metabase/redacted "db/details"})
```

Comment:
Not so sure about these values. I initially used strings like the password above, but this violated API schemas. I was not inclined to change the schema type, as this could a break. 

At the end of the day, having second thoughts about overloading the "type" of these fields - even when it's still a map like this, it still changes semantics and could cause confusing errors for an API client.

Reply:
Perhaps using empty maps are the least of all evils - although they don't give you anything to search for when debugging.